Slim/Control/Queries.pm

```diff
@@ -4524,7 +4524,7 @@ sub yearsQuery {
 	# get them all by default
 	my $where = {};
 
-	my ($key, $table) = ($hasAlbums || $libraryID) ? ('albums.year', 'albums') : ('id', 'years');
+	my ($key, $table) = ($hasAlbums || $libraryID) ? ('tracks.year', 'tracks') : ('id', 'years');
```

Comment:
The documentation says:

> hasAlbums:1	Return only years which are valid for albums.

Therefore IMHO we should keep the `$key` as is here.

Reply:
You are correct, I hadn't studied the code properly: ` ('tracks.year', 'tracks')` is of course the same as `('id', 'years')` as long as we're querying all libraries.

And yes, we should leave it to the client to tell us what it wants from the query via the `$hasAlbums` parameter. I'll introduce a new option to allow the user to toggle this. 

But I think the existing code would not display track-only years if `$hasAlbums=0` and `$libraryID` was defined. I'll attempt a fix.